packages/logger/README.md

```diff
@@ -0,0 +1,99 @@
+# Usage
+
+```js
+var Logger = require('@mongosh/logger')
+var path = require('path')
+var fs = require('fs')
+var os = require('os')
+
+var saveFilePath = path.join(os.homedir(), '.mongosh_log')
+var log = new Logger(saveFilePath)
+
+var infoObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+}
+
+var errorObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+  error: new TypeError()
+}
+
+log.info(infoObject)
+log.error(errorObject)
+```
+
+Logger uses [pino](https://github.com/pinojs/pino) under the hood for logging.
+
+Within mongosh architecture, `Logger` gets initialised along with the `Mapper`
+and requires passing in logger parameters to ShellMapper:
+
+![Using Logger with browser, Compass and CLI interfaces requires passing in
+logger parameters to ShellMapper](./logger-usage.png)
+
+```js
+var CliServiceProvider = require('mongosh-service-provider-server').CliServiceProvider
+var ShellApi = require('mongosh-shell-api')
+var Mapper = require('mongosh-mapper')
+
+var loggerPath = path.join(os.homedir(), '.mongosh_log')
+
+CliServiceProvider.connect(driverUri, driverOptions)
+  .then(function (serviceProvider) {
+    var mapper = new Mapper(serviceProvider, loggerPath)
```

Comment:
What if mapper would take a logger instance instead?

Reply:
that's also a possibility!

Do you feel like that'd be too many configurable things to do on the level of browser/Compass/CLI if now we have to configure the logger too? That was my biggest dilemma when thinking this through

Reply:
I guess I was mainly thinking, if someone was doing their own implementation of the whole thing, they might forget to setup a logger entirely and so no logging will ever happen. If that's something we want, then passing a Logger instance is a really good approach.

Reply:
I guess if we also let the implementer pass in the logger instance they can also decide on what logger to use, as long as it follows the `log.info`/ `log.error` pattern (most JS ones do). I guess it all depends on do we want to give implementers the choice?

Reply:
I prefer that form of dependency inversion as it makes it more testable (we can just do `new Mapper(serviceProviderMock, loggerMock))`.

It will take the creational responsibility out of the Mapper layer (the mapper won't need to know how to create a logger) allowing for independent evolution of the two implementations (as long as the interface is honored)

Reply:
yea i think what you're saying makes sense. let me have a think about it some more!